src/uproot/behaviors/TH1.py

```diff
@@ -14,6 +14,15 @@
 boost_axis_metadata = {"name": "fName", "label": "fTitle"}
 
 
+def _remove_nan_dims(array):
+    mask_axes = [
+        numpy.isnan(array).all(axis=tuple(numpy.delete(numpy.arange(array.ndim), i)))
+        for i in range(array.ndim)
+    ]
+    reshape_dim = [len(axis) - numpy.sum(axis) for axis in mask_axes]
+    return numpy.ma.masked_invalid(array).compressed().reshape(reshape_dim)
```

Comment:
Does this mean that everywhere Uproot used to return a NumPy array (`values`, `variances`, etc.), it now returns a NumPy MaskedArray?

MaskedArrays are not as compatible as plain NumPy arrays: more libraries are capable of working with `np.ndarray` than `np.ma.MaskedArray`. It would be better to fill the empty flow bins with either zeros or nans, but not `np.ma.masked`.

Reply:
No, `numpy.ma.masked_invalid(array).compressed()` is just another way of writing `array[~numpy.isnan(array)]`

Reply:
Okay, good. But then, doesn't this make the shape uncertain? What if there are legitimate nans among the bins?

Reply:
Yeah, this was a bit shaky, but I think it's reasonably robust now. Checks if all values along the "axis" are nan and then only removes these axes if at edges of the dimension. 

There's ambiguity if there's a real flow bin that is nan across all other axes, but I think that's the best we can if we don't have a way to store the actual info about which flow bins exists or not.

Reply:
The shape ambiguity remains as I mentioned in the opening comment, in case only overflow or underflow is set but not the other. 

We have 3 options
1. live with that ambiguity and have a matching return to `hist` (current PR)
2. return Nan/0 padded dimensions and break that convention
  - advantage is that it's clear* which flow bins existed in the original hist
  - disadvantage is breaking the `hist` convention and possibly some used code (probably not likely, because if anyone did use this with turned off flow bins, they'd raise this issue already)
3. return a masked array
  - advantage is making clear* which flow bins existed and being still somewhat consistent with the hist return shape-wise
  - disadvantage is returning a new type (probably breaking a bit more code than option 2), but I am not sure too many rely on `flow=True` in the first place

*clear - in the sense, I mentioned below, ambiguity if all flow bins are supposed to be nans, but I don't think there's a way around it

1 (current PR) is purely a fix without changing the current behaviour. The ambiguity already exists, but this option fixes the situation where `TH1.value()` doesn't return the real bin values at edges and  `TH1.value(flow=True)` returns edge bins instead of flow bins.

Reply:
I see what you were pointing out @jpivarski now. Fixed to remove dimensions by index instead, a test has been adapted to reflect that.